tests/flow/test_function_calls.py

```diff
@@ -2782,3 +2782,7 @@ def test94_vector(self):
         for q in queries_with_errors:
             self.expect_error(q, err_msg)
 
+    def test95_merge_map(self):
```

Comment:
This function needs to be relocated into `test_map.py`

The test as it is, is way too shallow, covering the bare basic of Map merging.

Extend it to cover a wide variety of cases.

e.g.
```
{a: b: {c:1 }} + {a: b: {c:1 }}
{a: b: {c:1 }} + {a: {b: {c:'1'}}}
{a: b: {c:1 }} + {a: {b: {d:3}}}
{a: b: {c:1 }} + {a: {b: {c:2, d:3}}}
```

Make sure all possibilities are covered.

Reply:
test moved
the cases I covered in one tests was:
1. key exists in a and not exists in b
2. key not exist in a and exists in b
3. key exists both on b and a

what cases is missing?